### ISSUE TITLE
Error: 1:1 'b' no such buffer appending '~quotes/to/filepath'

### ISSUE THREAD

**Issue by nonumeros** [mawww/kakoune#2897]
Go to `**debug**` buffer

     `:edit! ~/.config//kak/kakrc<ret>`

tab completion should lay out the working path e.g., `:e! /mypath/.config/kak/kakrc<ret>`

      `:b **debug**` 

as soon as `:b` is invoked thereafter to return to the file that was edited buffer, 

the errror `Error: 1:1: 'b' no such buffer '~.config/kak/kakrc'` is displayed and the file is unable to be accessed

scratch and debug buffers seem not affected

### Expected 

Switch buffers accordingly.

Deceiving behavior because it may not even happen the first time around but the second 

And it's the tilde :) for the home directory filesystem.

**Comment by nonumeros** [comment]
Can't believe this 

This only happens when it's not on the path environment of the installation 

so `/usr/local/bin/kak` and `/usr/bin/kak` are excluded from this behavior

**Comment by nonumeros** [comment]
One would think that if `buffer_cmd` is not doing what's supposed to , then at the very least `buffer_next_cmd` and `buffer_prev_cmd` should follow, but no, 

so the latter two recognize without a problem the directory path of the working file, while the former doesn't. 

But then again, a make install wouldnt' be affected by this. And That's the funny part after all. Not funny per se as if I were finding it laughable or comical but rather odd. 

the bottom line is that I have no problem when there are crashes and errors consistently, as long as this is a constant undisturbed by other elements. In this case if I were to `make install`, that alone does become the inconsistency.

**Comment by nonumeros** [comment]
As far as I can tell this problem goes as far back as https://github.com/mawww/kakoune/releases/tag/v2019.01.20 

I haven't had the time to read duplicate issues about this. No homework on my part. sorry. But as soon as `cd` ing onto another directory the error `Error: 1:1 no such buffer ` - which wasn't there before by switching back and forth between `debug` buffer and the file in question -  comes up.

**Comment by nonumeros** [comment]
closing this issue. Works as designed even though makes no sense
